lib/highLevelConsumer.js

```diff
@@ -523,11 +512,13 @@ HighLevelConsumer.prototype.offsetRequest = function (payloads, cb) {
  * @param {Client~failedToRegisterConsumerCallback} cb A function to call the consumer has been registered
  */
 HighLevelConsumer.prototype.registerConsumer = function (cb) {
-    this.client.zk.registerConsumer(this.options.groupId, this.id, this.payloads, function (err) {
+    var zk = this.client.zk;
+    var groupId = this.options.groupId;
+    this.client.zk.registerConsumer(groupId, this.id, this.payloads, function (err) {
         if (err) return cb(err);
+        zk.listConsumers(groupId, true);
```

Comment:
move `listConsumers` here is to ensure `consumersChanged` event after this consumer has registered to ZK. Otherwise `consumersChanged` event may be triggered twice.